issue_template: add template for performance issue (#9021)
pingcap/tidb: .github/ISSUE_TEMPLATE/performance-question.md

---
name: "\U0001F947 Performance Question"
about: Performance question about TiDB which is not caused by bug.

---

## Performance Question

- What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

- What's the difference between the performance you expect and the performance you observe?

- Are you comparing TiDB with some other database? What's the difference between TiDB and compared one?

- If it's a specific SQL query, could you please

>     - let us know whether you get your result after analyzing the table involved in the query.

Reviewer: Should we dump the stats before analyze, to save more information?

Author: Oh, here i mean whether he analyzed or not before finding the performance issue.
Is this sentence ambiguous?

Reviewer: I think so...

Author: changed

---
name: "\U0001F947 Performance Question"
about: Performance question about TiDB which is not caused by bug.

---

## Performance Question

> - What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

Reviewer 2: What do you think about use comment(HTML style) instread of brackets?

- What's the difference between the performance you expect and the performance you observe?

> - Are you comparing TiDB with some other database? What's the difference between TiDB and compared one?

Reviewer 3: ```suggestion
- Are you comparing TiDB with some other databases? What's the difference between TiDB and the compared one?
```

- If it's a specific SQL query, could you please
    - let us know whether you analyzed the tables involved in the query or not. And how long it is after you ran the last `ANALYZE`.
    - tell us whether this SQL always runs slowly, or just occasionally.
    - provide the `EXPLAIN ANALYZE` result of this query if you can wait until this SQL's execution finishes and your TiDB version is no lower than 2.1, or you can just provide the `EXPLAIN` result.
    - provide the plain text of the SQL and table schema so we can test it locally. It'll be more better if you can provide the dumped statistics information.
        - use `show create table ${involved_table}\G` to get the table schema.
        - use `curl -G "http://${tidb-server-ip}:${tidb-server-status-port}/stats/dump/${db_name}/${table_name}" > ${table_name}_stats.json` to get the dumped statistics of one involved table.
    - provide the `EXPLAIN` result of the compared database. For mysql, `EXPLAIN format=json`'s result will be more helpful.

>     - other information that you think it's useful to distinguish the property of this SQL.

Reviewer 3: ```suggestion
    - other information that is useful from your perspective.
```

- If it's not about a specific SQL query, e.g. the benchmark result you got by yourself is not expected. Could you please
    - show us your cluster's topology architecture

>     - a simple description of you workload

Author: I'not sure whether this two is enough

Reviewer: What about the metrics?

---
name: "\U0001F947 Performance Question"
about: Performance question about TiDB which is not caused by bug.

---

## Performance Question

- What version of TiDB are you using?
<!-- You can try `tidb-server -V` or run `select tidb_version();` on TiDB to get this information -->

> - What's the difference between the performance you expect and the performance you observe?

Reviewer 4: @zz-jason What's the observed and your expected performance respectively?

> - Are you comparing TiDB with some other databases? What's the difference between TiDB and the compared one?

Reviewer 4: ```suggestion
- Have you compared TiDB with other databases? If yes, what's their difference?
```

- If it's a specific SQL query, could you please

>     - let us know whether you analyzed the tables involved in the query or not. And how long it is after you ran the last `ANALYZE`.

Reviewer 5: s/how long it is/how long does the sql use ?

Author: No. It's <code>how long it takes from the time of your last `ANALYZE`</code>.

    - tell us whether this SQL always runs slowly, or just occasionally.

>     - provide the `EXPLAIN ANALYZE` result of this query if your TiDB version is higher than 2.1, or you can just provide the `EXPLAIN` result.

Reviewer 5: s/ higher than/ greater than or equal to 2.1.0

>     - provide the `EXPLAIN ANALYZE` result of this query if your TiDB version is higher than 2.1, or you can just provide the `EXPLAIN` result.

Reviewer 4: ```suggestion
    - The `EXPLAIN ANALYZE` result of this query if your TiDB version is higher than 2.1; otherwise, you can just provide the `EXPLAIN` result.
```

    - provide the plain text of the SQL and table schema so we can test it locally. It would be better if you can provide the dumped statistics information.
        <!-- you can use `show create table ${involved_table}\G` to get the table schema.-->
        <!-- use `curl -G "http://${tidb-server-ip}:${tidb-server-status-port}/stats/dump/${db_name}/${table_name}" > ${table_name}_stats.json` to get the dumped statistics of one involved table.-->
    - provide the `EXPLAIN` result of the compared database. For mysql, `EXPLAIN format=json`'s result will be more helpful.

>     - provide other information that is useful from your perspective.

Reviewer 4: ```suggestion
    - Other information that is useful from your perspective.
```

- If it's not about a specific SQL query, e.g. the benchmark result you got by yourself is not expected. Could you please
    - show us your cluster's topology architecture.
    - a simple description of you workload.

>     - the metrics PDF generated from grafana monitor. Remember that set the time range to when the performance issue happened.

Reviewer 4: ```suggestion
    - The metrics PDF generated from Grafana monitor. Remember to set the time range to the performance issue duration.
```
